static/app/components/integrationExternalMappingForm.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import capitalize from 'lodash/capitalize';
+import pick from 'lodash/pick';
+
+import {t, tct} from 'app/locale';
+import {ExternalActorMapping, Integration, Organization} from 'app/types';
+import {FieldFromConfig} from 'app/views/settings/components/forms';
+import Form from 'app/views/settings/components/forms/form';
+import {Field} from 'app/views/settings/components/forms/type';
+
+type Props = {
+  organization: Organization;
+  integration: Integration;
+  onSubmitSuccess: Form['props']['onSubmitSuccess'];
+  onCancel: Form['props']['onCancel'];
+  onSubmit?: Form['props']['onSubmit'];
+  mapping?: ExternalActorMapping;
+  sentryNames: {id: string; name: string}[];
+  type: 'user' | 'team';
+  baseEndpoint?: string;
+};
+
+export default class IntegrationExternalMappingForm extends React.Component<Props> {
+  get initialData() {
+    const {integration, mapping} = this.props;
+
+    return {
+      externalName: '',
+      userId: '',
+      teamId: '',
+      sentryName: '',
+      provider: integration.provider.key,
+      ...pick(mapping, ['externalName', 'userId', 'sentryName', 'teamId']),
+    };
+  }
+
+  get formFields(): Field[] {
+    const {sentryNames, type} = this.props;
+    const options = sentryNames.map(({name, id}) => ({value: id, label: name}));
+    const fields: any[] = [
+      {
+        name: 'externalName',
+        type: 'string',
+        required: true,
+        label: tct('External [type]', {type: capitalize(type)}),
+        placeholder: t(`${type === 'team' ? '@org/teamname' : '@username'}`),
+      },
+    ];
+    if (type === 'user') {
+      fields.push({
+        name: 'userId',
+        type: 'select',
+        required: true,
+        label: tct('Sentry [type]', {type: capitalize(type)}),
+        placeholder: t(`Choose your Sentry ${capitalize(type)}`),
+        options,
+        deprecatedSelectControl: false,
+      });
+    }
+    if (type === 'team') {
+      fields.push({
+        name: 'teamId',
+        type: 'select',
+        required: true,
+        label: tct('Sentry [type]', {type: capitalize(type)}),
+        placeholder: t(`Sentry ${capitalize(type)}`),
```

Comment:
@NisanthanNanthakumar this isn't going to translate properly, you should use `tct` like you do with the label